framework/src/node/consensus/network_endpoint.ts

```diff
@@ -219,8 +225,12 @@ export class NetworkEndpoint {
 			throw error;
 		}
 
+		let isValidCommit: boolean;
 		try {
-			this._commitPool.validateCommit(decodedData.singleCommit);
+			isValidCommit = await this._commitPool.validateCommit(
+				this._apiContext,
+				decodedData.singleCommit,
+			);
```

Comment:
```suggestion
			const isValidCommit = await this._commitPool.validateCommit(
				this._apiContext,
				decodedData.singleCommit,
			);
			if (!isValidCommit) { return }
```

Reply:
you can return within try block only

Reply:
We need Ishan as a Service as the world, amazing as always 😄 Will do.

Reply:
done